src/main.rs

```diff
@@ -949,6 +1037,77 @@ async fn main() {
         );
     }
 
+    tokio::spawn(async {
+        let local_port = match env::var("LAVA_GITLAB_RUNNER_LOCAL_PORT") {
+            Ok(val) => val
+                .parse()
+                .expect("failed to parse LAVA_GITLAB_RUNNER_LOCAL_PORT as a port number"),
+            Err(_) => {
+                warn!("No LAVA_GITLAB_RUNNER_LOCAL_PORT set, will listen on ephemeral IP.");
+                0u16
+            }
+        };
+
+        let listener = std::net::TcpListener::bind(std::net::SocketAddr::new(
+            IpAddr::V4(Ipv4Addr::UNSPECIFIED),
+            local_port,
+        ))
+        .expect("failed to bind listener");
+
+        let routable_host = match env::var("LAVA_GITLAB_RUNNER_ROUTABLE_HOST") {
+            Ok(val) => val,
+            Err(_) => {
+                let host = local_ip_address::local_ip()
+                    .expect("failed to determine local ip")
+                    .to_string();
+
+                warn!(
+                    "No LAVA_GITLAB_RUNNER_ROUTABLE_HOST set, using best guess of local IP {}.",
```

Comment:
If there is no base url set then simply don't enable the listen functionality; local_pi() is of quite questionable value.